pages/api/user/index.ts

```diff
@@ -19,7 +25,26 @@ export default async function handle(
             break;
         }
         case "POST": {
-            // TODO:
+            let newUser;
```

Comment:
Will you guys be adding a validation function for POST user? One thing that we have to ensure is that the roles are mutually exclusive. A user can't be a teacher and a volunteer. Before we create a role, we should make sure that they aren't assigned any role.

Reply:
Perhaps to make it more easy to determine what role the user is, we should have a `role` field in the user table where the value is an ENUM with values (TEACHER, ADMIN, VOLUNTEER, PARENT). This way we don't have to join with so many tables just to find what the user is. Then, in `POST`, we check that is `user.role` is NULL, then we allow the create to occur. If is it not NULL, we return an error as the user is assigned a role (has a record in the corresponding associating table). 

cc: @cindy-wang328 @mahadzaryab1 Let me know what you guys think about this

Reply:
https://github.com/uwblueprint/social-diversity-for-children/pull/18/commits/db29fdec0e99932472e54046dae54f53be52fcec